src/NuGet.Core/NuGet.LibraryModel/LibraryDependency.cs

```diff
@@ -28,6 +28,11 @@ public class LibraryDependency : IEquatable<LibraryDependency>
         /// </summary>
         public bool AutoReferenced { get; set; }
 
+        /// <summary>
+        /// True if the dependency has the version set through CentralPackagVersionManagement file.
+        /// </summary>
+        public bool VersionCentrallyManaged { get; set; }
```

Comment:
Apologies if this has already been discussed, but the only place in production code this seems to be read is when checking if NU1008 needs to be raised or not.  Could the check for NU1008 happen before CPVM versions are applied to PackageReferences, so that this bool is not needed?